coremltools/models/model.py

```diff
@@ -602,3 +615,40 @@ def _verify_input_name_exists(self, input_dict):
                 err_msg = "Provided key \"{}\", in the input dict, " \
                           "does not match any of the model input name(s), which are: {}"
                 raise KeyError(err_msg.format(given_input, ",".join(model_input_names)))
+
+    _supported_dtype = [
+        numpy.float32,
+        numpy.float64,
+        numpy.int32,
+    ]
+
+    def _verify_input_type(self, input_dict):
+        model_input_types = {}
+        for inp in self._spec.description.input:
+            type_value = inp.type.multiArrayType.dataType
+            type_name = inp.type.multiArrayType.ArrayDataType.Name(type_value)
+            if type_name != "INVALID_ARRAY_DATA_TYPE":
+                model_input_types[inp.name] = type_name
+
+        for given_input_name, given_input in input_dict.items():
+            if not given_input_name in model_input_types:
+                continue
+            input_dict[given_input_name] = self._convert_to_numpy_if_enable(given_input, given_input_name)
+            given_input = input_dict[given_input_name]
+            if given_input.dtype not in self._supported_dtype:
```

Comment:
Other numpy types may actually be supported, I'm not sure. Let's just change this line to:
```python
if no isinstance(given_input, numpy.ndarray):
```

Reply:
I confirm code below was failed with `RuntimeError`.

```python
import torch
import coremltools as ct
import numpy

class FlattenWithReshape(torch.nn.Module):
    def forward(self, x):
        s1 = x.shape[1] * x.shape[2]
        return x.reshape((x.shape[0], s1))  # => crash with different input shape

model = FlattenWithReshape()
x = torch.randn((3, 2, 6))
traced_model = torch.jit.trace(model, (x,))
in_shape = ct.Shape(shape=(ct.RangeDim(1, 30), ct.RangeDim(1, 30), ct.RangeDim(1, 30)), default=x.shape)
model_ct = ct.convert(traced_model, inputs=[ct.TensorType(shape=in_shape, name='x')])

out = traced_model(x)
out_dict = model_ct.predict({'x': x.detach().numpy().astype(numpy.int64)})  # passed only when float32, float64, int64
```

However, your suggestion looks good in situations where the specifications are unclear.